### ISSUE TITLE
[BUG] CMake issue with clang-format on ubuntu 22.04

### ISSUE THREAD

**Issue by robeme** [lammps/lammps#3484]
It seems that CMake is not able to identify the correct clang-format version if the output of `clang-format --version` is something like `Ubuntu clang-format version 14.0.0-1ubuntu1` using 
```
    string(REGEX REPLACE "clang-format version ([0-9.]+).*"
                         "\\1"
                         ClangFormat_VERSION
                         "${clang_format_version}")
```
in `cmake/Modules/FindClangFormat.cmake`. I only get the message `Could NOT find ClangFormat: Found unsuitable version "0.0", but required is at least "8.0" (found /usr/bin/clang-format)`. Unfortunately, I don't know too much of regex nor CMake to modify it accordingly. Maybe this is also an issue with other libraries and packages?

**Comment by akohlmey** [comment]
Thanks for reporting. I will take a look. In the meantime, you can just run `clang-format -i` interactively on relevant files or folders and it should pick up the embedded configuration and apply it.

**Comment by akohlmey** [comment]
This is now fixed in commit c741eca953ab58fd170e1896c938c54c73cd51d6 and will be available with the next patch release of LAMMPS

**Comment by robeme** [comment]
Thanks! That was quick. It's kinda funny that a tool which helps to consistently format source code does somehow not consistently output its version number 🤣...

**Comment by akohlmey** [comment]
> Thanks! That was quick. It's kinda funny that a tool which helps to consistently format source code does somehow not consistently output its version number rofl...

I think this is a modification from the Ubuntu developers. On Fedora and Arch Linux it just uses the output starting with `clang-format`.

**Comment by bestucan** [comment]
> This is now fixed in commit [c741eca](https://github.com/lammps/lammps/commit/c741eca953ab58fd170e1896c938c54c73cd51d6) and will be available with the next patch release of LAMMPS

This modification also work on debian 12. Just replace "Ubuntu" with "Debian"

**Comment by akohlmey** [comment]
> This modification also work on debian 12. Just replace "Ubuntu" with "Debian"

@bestucan thanks for the note. will update the CMake script accordingly.
